app/src/main/java/com/github/se/eventradar/ui/home/Home.kt

```diff
@@ -199,6 +262,193 @@ fun HomeScreen(
   }
 }
 
+@Composable
+fun SearchBarAndFilter(
+    searchQuery: String,
+    onSearchQueryChange: (String) -> Unit,
+    onSearchButtonClick: () -> Unit,
+    showFilterPopUp: Boolean,
+    setShowFilterPopUp: (Boolean) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Row(
+        verticalAlignment = Alignment.CenterVertically,
+        modifier = modifier
+    ) {
+        // Search bar
+        TextField(
+            value = searchQuery,
+            onValueChange = { onSearchQueryChange(it) },
```

Comment:
This doesn't seem to be working. When I type in the search bar nothing happens. You most likely will have to modify this in the view model to change the actual UI state (e.g. `_uiState`). See the bootcamp for an example.

Reply:
Sorry - I forgot to push the latest version. This will be updated in a separate pull request to implement functionality via the VM.